### PR TITLE
Add transmittable domain model 185007548

### PR DESCRIPTION
**Pull request by ymhari**

### CODE REVIEW

**Review by adamrunner — DISMISSED** [review]

File: lib/aca_entities/protocols/transmittable/transaction.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        # @!attribute [r] timestamps
+        # Date and time that this resource was originally created and last updated. Time is represented in
+        # Coordinated Univeral Time (UTC). System managed.
+        # @return [AcaEntities::TimeStamp]
+        attribute :timestamps, AcaEntities::TimeStamp.optional.meta(omittable: true)
+
+        # @!attribute [r] json_payload
+        # An optional field that stores a json payload being sent as part of the transmission
+        # @return [String]
+        attribute :json_payload, Types::Hash.meta(omittable: true)
```
Comment (polographer) [inline lib/aca_entities/protocols/transmittable/transaction.rb:58]: I know that the focus is on newer systems, but we have a couple of services that uses xml (ie, rrv,pvc), could it be better to name it just :payload ?

File: spec/aca_entities/protocols/transmittable/contracts/process_status_contract_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe AcaEntities::Protocols::Transmittable::Contracts::ProcessStatusContract do
+  subject { described_class.new }
+
+  context 'Given valid required params' do
```
Comment (polographer) [inline spec/aca_entities/protocols/transmittable/contracts/process_status_contract_spec.rb:8]: I think this spec is just a stub

File: spec/aca_entities/protocols/transmittable/contracts/process_state_contract_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe AcaEntities::Protocols::Transmittable::Contracts::ProcessStateContract do
```
Comment (polographer) [inline spec/aca_entities/protocols/transmittable/contracts/process_state_contract_spec.rb:5]: I think these file is just a stub

File: lib/aca_entities/protocols/transmittable/process_status.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module AcaEntities
+  module Protocols
+    module Transmittable
+      # The current state of a process and the history of its state transitions
+      class ProcessStatus < Dry::Struct
```
Comment (polographer) [inline lib/aca_entities/protocols/transmittable/process_status.rb:7]: this is a question; this looks slightly similar to workflow_state_transition.rb, but the other one is more targeted for state machine, is the purpose here to don't create a state machine ?

Also I may be misunderstanding the whole thing, but the comment I'm going is why not re-use the transitions object, or maybe the purpose is to reflect a state in time and not hold all the states

Comment (kristinmerbach) [reply lib/aca_entities/protocols/transmittable/process_status.rb:7]: We do want all the states for history tracking, while process status is the current view with the history available. This was a pattern designed by Dan that we implemented.

**Review by polographer — APPROVED** [review]

**Review by adamrunner — APPROVED** [review]

**Pull request merged**